src/dag-node/util.js

```diff
@@ -0,0 +1,36 @@
+'use strict'
+
+const DAGLink = require('./../dag-link')
+
+exports = module.exports
+
+function cloneData (dagNode) {
+  let data = new Buffer(0)
+
+  if (dagNode.data && dagNode.data.length > 0) {
+    data = new Buffer(dagNode.data.length)
+    dagNode.data.copy(data)
+  }
+
+  return data
+}
```

Comment:
this can be simplified to 

```js
function cloneData (dagNode) {
  assert(Buffer.isBuffer(dagNode.data), 'Not a valid dagnode')
  
  const data = new Buffer(dagNode.data.length)
  dagNode.data.copy(data)
  return data
}
```